pandas/core/indexes/multi.py

```diff
@@ -1095,7 +1094,8 @@ def _format_native_types(self, na_rep="nan", **kwargs):
             if mask.any():
                 nan_index = len(level)
                 level = np.append(level, na_rep)
-                level_codes = level_codes.values()
+                assert not level_codes.flags.writeable  # i.e. copy is needed
```

Comment:
shoulnt' you be calling _coerce_indexer_frozen here (with copy=True)?

Reply:
no, _coerce_indexer_frozen returns an ndarrary that is not writeable, whereas here we have a non-writeable ndarray and need to get a writeable copy